### DISCUSSION TITLE
Escaping a comma in --driver-opt

### DISCUSSION THREAD

**Discussion by juergenthomann** [docker/buildx#1133] [Q&A] [answered]
Hi

I'm trying to set a no_proxy variable with multiple comma separated values for buildx, but I'm unable to pass them to buildx without it complaining about the syntax. I also tried different ways to quote it but without success.

> $ docker buildx create --use --driver-opt env.no_proxy='localhost,127.0.0.1'
error: invalid value "127.0.0.1", expecting k=v

Is there any way to escape it? I tried backslash as well before the comma without any difference.

**Comment by jedevc** [comment, marked as answer]
Sorry for the late reply, but I just had a similar question myself :smile: 

I think you should be able to escape by enclosing the key **and** the value in quotes, i.e. something like:
```
$ docker buildx create --use --driver-opt \'env.no_proxy=localhost,127.0.0.1\',adifferentopt=value
```

You need the backslashes to escape the quotes from bash so buildx actually gets them, and then you each key/value pair should be kept separate :tada:

**Reply by juergenthomann** [reply]
I tried the following command, but even with this style of escaping it parses it wrong and same error for bash and sh

```
$ docker buildx create --use --name test --driver-opt \'env.no_proxy=localhost,127.0.0.1\'
error: invalid value "127.0.0.1'", expecting k=v
```

With strace I can confirm, that docker is called with single quotes. I also tried it with just quoting the value and escaping it for bash, but same error message

```
execve("/usr/bin/docker", ["docker", "buildx", "create", "--use", "--name", "test", "--driver-opt", "'env.no_proxy=localhost,127.0.0.1'"]
```

**Reply by jedevc** [reply]
Right! Sorry, this is what I get for answering before having coffee :facepalm: 

You need to use double quotes, not single quotes, my bad! So something like *should* actually work:
```
> $ docker buildx create --use --driver-opt \"env.no_proxy=localhost,127.0.0.1\",adifferentopt=value
```

**Reply by juergenthomann** [reply]
Thank you, I can confirm that escaped double quotes are working.

**Comment by snowdream** [comment]
I tried as follows and fix it: 
https://github.com/docker/setup-buildx-action/issues/57#issuecomment-1059657292
